### PR TITLE
Hotfix/create endturn feature

### PR DESCRIPTION
**Pull request by KENUBBE**
Hotfix for createEndturnFeature() 
* Added opponent player variable to track currentPlayer and opponentPlayer

### CODE REVIEW

File: src/main/java/models/GameEngine.java [hunk 1]
```diff
@@ -11,15 +12,17 @@ public GameEngine(){
...
 
     private Player p1, p2;
     private ArrayList<Card> gameCards;
     private Player currentPlayer;
-    private Player opponent;
+    private Player opponentPlayer;
     private Deck deck;
     private boolean game;
     private boolean playing;
+    private Random rand;
```
Comment (Juulis) [inline src/main/java/models/GameEngine.java]: Finns det någon anledning att rand är en klassvariabel?

File: src/main/java/models/GameEngine.java [hunk 2]
```diff
@@ -77,26 +80,33 @@ public void initPlayer() {
...
         if (randomNr == 1) {
             currentPlayer = p1;
+            opponentPlayer = p2;
         } else {
             currentPlayer = p2;
+            opponentPlayer = p1;
         }
     }
 
+    public Player getOpponentPlayer(){
```
Comment (Juulis) [inline src/main/java/models/GameEngine.java:132]: 🤒 😷     🔥 🔥 🔥 

😆 🤣

File: src/main/java/models/GameEngine.java (hunk 1 above)
Comment (KENUBBE) [reply src/main/java/models/GameEngine.java]: Because I can!!!

Nä men absolut, den tar upp onödigt minne i heapen😝I fix for you!

**Comment by KENUBBE** [comment]
* Rewrote old tests for endTurn() and getStartingPlayer() to use spy instead. Because of that I was able to remove "int randomNr" as an argument from getStartingPlayer().

* Added randomizer as a dependency instead of a instance variable, so that we can use it wherever we need to.

File: src/main/java/models/GameEngine.java (hunk 2 above)
Comment (mohammedalmahfoodh) [reply src/main/java/models/GameEngine.java:132]: I think you missed range values[1,2] int randomNr = rand.nextInt();

Comment (Juulis) [reply src/main/java/models/GameEngine.java:132]: mohammed har rätt, range saknas! :P

File: src/main/java/models/GameEngine.java [hunk 3]
```diff
@@ -77,26 +78,38 @@ public void initPlayer() {
 
     /**
      * Determines if p1 or p2 is going to start
-     *
-     * @param randomNr
      */
-    public void getPlayerToStart(int randomNr) {
+    public void getPlayerToStart() {
+        Random rand = makeRandom();
+        int randomNr = rand.nextInt();
```
Comment (Juulis) [inline src/main/java/models/GameEngine.java]: what range?

**Pull request merged**